Services/sdLitica.AnalyticsManagementCore/AnalyticsRegistry.cs

```diff
@@ -0,0 +1,161 @@
+        }
+
+        /// <summary>
+        /// Register analytical module
+        /// </summary>
+        /// <param name="module"></param>
+        public void Register(AnalyticsModuleRegistrationModel module)
+        {
+            CheckAvailable();
+            if (_analyticsModuleRepository.Contains(module.ModuleGuid))
+            {
+                AnalyticsModule analyticsModule_ = _analyticsModuleRepository.GetById(module.ModuleGuid);
+                analyticsModule_.LastHeardTime = DateTime.Now;
+            }
+            else
+            {
+                _analyticsModuleRepository.Add(new AnalyticsModule()
+                {
+                    Id = module.ModuleGuid,
+                    LastHeardTime = DateTime.Now,
+                    QueueName = module.QueueName
+                });
+            }
+            _analyticsModuleRepository.SaveChanges();
+
+            AnalyticsModule analyticsModule = _analyticsModuleRepository.GetById(module.ModuleGuid);
+            foreach (AnalyticsOperationModel operationModel in module.Operations) {
+                if (_analyticsOperationRepository.ContainsName(operationModel.Name))
+                {
+                    AnalyticsOperation analyticsOperation = _analyticsOperationRepository.GetByName(operationModel.Name);
+                    
+                    if (!analyticsOperation.ModulesOperations.Select(mo => mo.AnalyticsModule).Contains(analyticsModule)) {
+                        ModulesOperations moduleOperation = new ModulesOperations()
+                        {
+                            AnalyticsOperation = analyticsOperation,
+                            AnalyticsModule = analyticsModule
+                        };
+                        _modulesOperationsRepository.Add(moduleOperation);
+                        analyticsModule.ModulesOperations.Add(moduleOperation);
+                        analyticsOperation.ModulesOperations.Add(moduleOperation);
+
+                    }
+                }
+                else
+                {
+                    AnalyticsOperation operation = new AnalyticsOperation()
+                    {
+                        Id = Guid.NewGuid(),
+                        Name = operationModel.Name,
+                        Description = operationModel.Description
+                    };
+                    ModulesOperations moduleOperation = new ModulesOperations()
+
+                    {
+                        AnalyticsOperation = operation,
+                        AnalyticsModule = analyticsModule
+                    };
+                    _modulesOperationsRepository.Add(moduleOperation);
+                    analyticsModule.ModulesOperations.Add(moduleOperation);
+                    operation.ModulesOperations.Add(moduleOperation);
+
+                    _analyticsOperationRepository.Add(operation);
+
+                }
+            }
+            _analyticsOperationRepository.SaveChanges();
```

Comment:
I am not sure, but I think that the first SaveChanges will flush entire Context.

Reply:
true